lib/validators/at_least_one_group_validator.rb

```diff
@@ -3,13 +3,25 @@
 class AtLeastOneGroupValidator
   def initialize(opts = {})
     @opts = opts
+    @invalid_groups = []
   end
 
   def valid_value?(val)
-    val.present? && val != ""
+    @invalid_groups = []
```

Comment:
A bit naughty. To be able to carry the state properly, the validator should take the value as part of initialization, but that change requires quite a bit of rewiring of the type supervisor. So for now I reset the state here in case the validator is reused, which I'm pretty sure it isn't, but this is technically correct just in case.